### PR TITLE
Refactor courses crawler from former version

### PR DESCRIPTION
**Pull request by Jhuni0123**
`course_model.md` defines course entity.
`crawler.py` crawls courses in some years you want.

### CODE REVIEW

File: backend/course/crawler.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        'U000200001U000300001': '1',
+        'U000200001U000300002': 'S',
+        'U000200002U000300001': '2',
+        'U000200002U000300002': 'W'
+    }
+    # srchOpenSchyy : year
+    # srchOpenShtm : semester
+    # srchOpenUpSbjtFldCd : area
+    # srchOpenSbjtFldCd : subarea
+    form_data = {
```
Comment (dj-shin) [inline backend/course/crawler.py]: This `form_data` isn't needed in crawling logic and unnecessarily long. I prefer making new `course/format.py` file and putting `form_data` there.

File: backend/course/crawler.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    def get_courses(self, year, semester, area, subarea):
+        form_data = self.form_data.copy()
+        form_data['srchOpenSchyy'] = str(year),
+        form_data['srchOpenShtm'] = semester,
+        form_data['srchOpenUpSbjtFldCd'] = area['code'],
+        form_data['srchOpenSbjtFldCd'] = subarea['code'],
+        excel = requests.post(self.excel_url, form_data)
+        try:
+            workbook = xlrd.open_workbook(file_contents=excel.content)
+        except Exception as e:
```
Comment (dj-shin) [inline backend/course/crawler.py]: `Exception` should be more concrete.

**Review by dj-shin — CHANGES_REQUESTED** [review]
Minor changes recommended.

File: backend/course/crawler.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        'srchOpenSbjtTmVal': '',
+        'srchLsnProgType': '',
+        'srchMrksGvMthd': '',
+        'srchFlag': '',
+    }
+
+    def parse_option_tag(self, op):
+            return {'code': op['value'], 'name': op.text.strip()}
+
+    def get_areas(self, year, semester):
```
Comment (sonowz) [inline backend/course/crawler.py:20]: It seems that `get_areas()`, `get_subareas()`, `get_courses()` functions have quite redundant code.
Would there be any way to make it more simple?

File: backend/course/crawler.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
+import requests
+from bs4 import BeautifulSoup as bs
+import xlrd
```
Comment (skystar-p) [inline backend/course/crawler.py]: these modules should be included in `requirements.txt`.

Comment (skystar-p) [inline backend/course/crawler.py:3]: these modules should be included in `requirements.txt`.

**Review by skystar-p — COMMENTED** [review]
prerequisites update is recommended.

**Review by dj-shin — APPROVED** [review]

**Comment by dj-shin** [comment]
@skystar-p @dnsdhrj Thank you for reviews :smiley:

**Pull request merged**